src/github.rs

```diff
@@ -139,29 +142,58 @@ pub fn validate_pull_request(job: &worker::PullRequestJob, client: &Github) -> R
     Ok(failures)
 }
 
+pub trait TryExecute: Executor {
+    fn try_execute<T: DeserializeOwned>(self) -> Result<T>
+    where
+        Self: Sized
+    {
+        #[derive(Deserialize)]
+        struct GithubError {
+            message: String,
+        }
+
+        (|| match self.execute::<serde_json::Value>() {
```

Comment:
why is this a lambda? it's getting called immediately, I don't see a reason why this wouldn't just be inlined.